packages/lib/modules/tokens/approvals/permit2/permit2.helpers.ts

```diff
@@ -1,64 +1,94 @@
 
   const approvalExpired = (tokenAddress: Address) =>
-    expirations[tokenAddress] >= getNowTimestampInSecs()
-  const alreadyAllowed = (amountIn: TokenAmount) =>
-    !approvalExpired(amountIn.token.address) &&
-    allowedAmounts[amountIn.token.address] >= amountIn.amount
-  const amountInValid = (amountIn: TokenAmount) =>
+    expirations[tokenAddress] < getNowTimestampInSecs()
+  const alreadyAllowed = (amountIn: TokenAmountIn) =>
+    !approvalExpired(amountIn.address) && allowedAmounts[amountIn.address] >= amountIn.amount
+  const amountInValid = (amountIn: TokenAmountIn) =>
     amountIn.amount === 0n || alreadyAllowed(amountIn)
-  const isValid = !!queryOutput?.sdkQueryOutput.amountsIn.every(amountInValid)
+  const isValid = tokenAmountsIn.every(amountInValid)
+  /*
+   // Delete after debug:
+   if (tokenAmountsIn) {
+     const tokenAddress = tokenAmountsIn[0].address
+     const amountIn = tokenAmountsIn[0]
+     console.log({tokenAddress,
+       approvalExpired: approvalExpired(tokenAddress),
+       alreadyAllowed: alreadyAllowed(amountIn),
+     })
+   }
+  */
   return isValid
 }
 
 type BasePermit2Params = {
-  queryOutput?: SdkQueryAddLiquidityOutput
+  tokenAmountsIn?: TokenAmountIn[]
+  chainId: number
   wethIsEth: boolean
 }
 
 // Returns the symbols of the tokens that need to be approved for permit2
 export function getTokenSymbolsForPermit2({
   getToken,
-  queryOutput,
+  tokenAmountsIn,
   wethIsEth,
-}: BasePermit2Params & { getToken: GetTokenFn }) {
-  if (!queryOutput) return []
-  const chain = getGqlChain(queryOutput.sdkQueryOutput.chainId)
+  chainId,
+}: BasePermit2Params & { getToken: GetTokenFn }): string[] {
+  if (!tokenAmountsIn) return []
+  const chain = getGqlChain(chainId)
   const tokenSymbols = filterWrappedNativeAsset({
     wethIsEth,
-    amountsIn: queryOutput.sdkQueryOutput.amountsIn,
+    tokenAmountsIn,
     chain,
   })
-    .filter(a => a.amount > 0n)
-    .map(a => getToken(a.token.address, chain)?.symbol ?? 'Unknown')
+    .filter(t => t.amount > 0n)
+    .map(t => getToken(t.address, chain)?.symbol ?? 'Unknown')
   return tokenSymbols
 }
 
 // Returns the token addresses that need to be approved for permit2
 export function getTokenAddressesForPermit2({
   wethIsEth,
-  queryOutput,
+  tokenAmountsIn,
+  chainId,
 }: BasePermit2Params): Address[] | undefined {
-  if (!queryOutput?.sdkQueryOutput) return undefined
-  const chain = getGqlChain(queryOutput.sdkQueryOutput.chainId)
-  const result = filterWrappedNativeAsset({
+  if (!tokenAmountsIn) return undefined
+  const chain = getGqlChain(chainId)
+  return filterWrappedNativeAsset({
     wethIsEth,
     chain,
-    amountsIn: queryOutput.sdkQueryOutput.amountsIn,
-  }).map(a => a.token.address)
-  return result
+    tokenAmountsIn,
+  }).map(t => t.address)
+}
+
+export function permit2Address(chain: GqlChain): Address {
+  return getNetworkConfig(chain).contracts.permit2 || ('' as Address)
+}
```

Comment:
Should this really return an empty address ever?

Reply:
Added a TODO comment before that line:
`  // TODO: Remove ('' as Address) when all chains have permit2 defined to it's not optional anymore`